### ISSUE TITLE
multi channel data processing

### ISSUE THREAD

**Issue by myy-imp** [MIC-DKFZ/nnUNet#847]
Hi Isensee,

         Thanks for your work！I just want to change multi-channel 3D UNet to multi-channel multi path 3D UNet，as the picture show. Each channel deals with one modality. I want to know what I need to pay attention to when I modify the code. Do i need to modify the dataset.json file?  Is it affected when determining the patch size?
           
![image](https://user-images.githubusercontent.com/42831398/141256655-20651e00-841f-4679-a9be-0bff89f59425.png)

**Comment by FabianIsensee** [comment]
It is best to just run plan and preprocess as normal. Then you build your own trainer class where you overwrite the https://github.com/MIC-DKFZ/nnUNet/blob/96d44c2fc1ce5e18da4cd54bf52882047c37982e/nnunet/training/network_training/nnUNetTrainerV2.py#L129 function to use your own network. All you need to make sure is that your network can use the patch size nnU-Net wants to use and that it will all fit into your GPU memory

**Comment by myy-imp** [comment]
Thanks for your reply！I have another question，Does the training result below seem normal? Why is it so volatile? If it is not normal, what are the possible reasons and what should I do to make it better? Thank you!
![捕获](https://user-images.githubusercontent.com/42831398/145957574-ed64da84-08dd-4349-9a34-c090fd9e3031.JPG)

**Comment by FabianIsensee** [comment]
Hi,
the first thing you should do is run a normal nnU-Net training (no modifications). Use this as a baseline. If you change something and it becomes volatile then it's hard to guess what might be going on. Also you really should run that baseline to make sure you are actually improving things!
Depending in the dataset some volatility is to be expected. The training curve you posed here looks strange though. What dataset is this?

**Comment by myy-imp** [comment]
Hi， I used 4DCT and ran the normal nnU-Net 2d training (no modifications).   I used the  trained model (the training results are shown in the figure above) to predict and evaluate the test data, the mean dice was 0.2. Then I used the trained model to predict the training data, the mean dice was 0.3. There must be something wrong.  But I have no idea where is wrong. I did not do any modifications.

**Comment by FabianIsensee** [comment]
Was there a specific reason to use the 2D model? Have you tried the 3d models as well?
It's very hard to say what's going on without looking into the data. You can not share it can you"?

**Comment by myy-imp** [comment]
Can I just give you a few examples？How can I give it to you？

**Comment by FabianIsensee** [comment]
Best is to send via email (or send a download link via email): f.isensee@dkfz.de

**Comment by myy-imp** [comment]
Hi Fabian, I have checked the whole code, and found that self.num_batches_per_epoch  was set to  2. This should be the reason for the bad result. Then I retrained the 3d_fullres model and got the result shown below. It seems overfitting. Is there any solution to solve it? The number of training data is 40. I still sent you some of my data. 
![image](https://user-images.githubusercontent.com/42831398/146860496-8affe888-7a2d-4b4c-9191-c06b37bc265b.png)
